frontend/src/components/GManageShootCustomFields.vue

```diff
@@ -503,7 +517,38 @@ async function addField () {
   } else {
     addShootCustomField(editedField.value)
   }
-  closeDialog()
+
+  return updateConfiguration(closeDialog)
+}
+
+function deleteField () {
+  deleteShootCustomField(deleteFieldItem.value)
+  return updateConfiguration(closeDeleteDialog)
+}
+
+async function updateConfiguration (onSuccess) {
+  loading.value = true
+  try {
+    const { metadata: { name, namespace } } = projectStore.project
+    const mergePatchDocument = {
+      metadata: { name, namespace },
+    }
+    Object.assign(mergePatchDocument, getShootCustomFieldsPatchData())
+
+    await projectStore.patchProject(mergePatchDocument)
```

Comment:
The patchData function should be a computed property that contains also the name and namespace:
```js
  const shootCustomFieldsPatchData = computed(() => {
    const { namespace, name } = projectItem.metadata
    return {
      metadata: {
        namespace,
        name,
        annotations: {
          'dashboard.gardener.cloud/shootCustomFields': getProjectAnnotation('dashboard.gardener.cloud/shootCustomFields', null),
        },
      },
    }
  })
```